plugins/usertiming.js

```diff
@@ -0,0 +1,110 @@
+/**
+\file usertiming.js
+Plugin to collect metrics from the W3C User Timing API.
+For more information about User Timing,
+see: http://www.w3.org/TR/user-timing/
+
+This plugin is dependent on the UserTimingCompression library
+see: https://github.com/nicjansma/usertiming-compression.js
+UserTimingCompression must be loaded before this plugin's init is called.
+*/
+
+(function() {
+
+	BOOMR = BOOMR || {};
+	BOOMR.plugins = BOOMR.plugins || {};
+
+	if (BOOMR.plugins.UserTiming) {
+		return;
+	}
+
+	/**
+	 * @returns String compressed user timing data that occurred since the last call
+	 */
+	function getUserTiming() {
+		var map, res, now = BOOMR.now();
+
+		map = UserTimingCompression.getCompressedUserTiming(impl.options);
+		res = UserTimingCompression.compressForUri(map);
+		impl.options["from"] = now;
```

Comment:
It will be hard to avoid duplicates and not skip some measures without filtering by  `startTime + duration` in `UserTimingCompression`.
The alternative is to call `clearMarks` and `clearMeasures` on beacon, but that's obtrusive to other js on the page that would need the timing entries.